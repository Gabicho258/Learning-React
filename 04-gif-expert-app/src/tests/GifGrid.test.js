

import React from 'react';
import {shallow} from 'enzyme'
import { GifGrid } from '../components/GifGrid';
import { useFetchGifs } from '../hooks/useFetchGifs';
jest.mock("../hooks/useFetchGifs")

describe('Pruebas en GifGrid', () => {

    // test('debe hacer match con el snapshot', () => {
    //     const wrapper = shallow(<GifGrid category = "gtav"/>)
    //     expect(wrapper).toMatchSnapshot();
    // })
    test('debe mostrar items cuando se cargan imagenes useFetc', () => {
        const gifs = [{
            id : "ABC",
            url : "http://www.w3.org/1999/",
            title : "titulo"
        },
        {
            id : "123",
            url : "http://www.w3.org/1999/",
            title : "titulo"
        }]
        useFetchGifs.mockReturnValue({
            data : gifs,
            loading: false
        })
        const wrapper = shallow(<GifGrid category = "gtav"/>)


        // expect(wrapper.find("p").exists).toBe(false)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
    })
    
})
