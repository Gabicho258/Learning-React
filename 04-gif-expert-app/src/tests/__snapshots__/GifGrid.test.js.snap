// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid debe mostrar items cuando se cargan imagenes useFetc 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    gtav
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="titulo"
      url="http://www.w3.org/1999/"
    />
    <GifGridItem
      id="123"
      key="123"
      title="titulo"
      url="http://www.w3.org/1999/"
    />
  </div>
</Fragment>
`;
